### ISSUE TITLE
🐛 BUG: Using @rollup/plugin-dsv plugin returns incorrect result

### ISSUE THREAD

**Issue by Simperfy** [withastro/astro#2237]
### What version of `astro` are you using?

0.21.13

### What package manager are you using?

yarn

### What operating system are you using?

Linux (Ubuntu 20.04)

### Describe the Bug

When trying to use [@rollup/plugin-dsv](https://www.npmjs.com/package/@rollup/plugin-dsv) the results are incorrect it includes "export default ..." *(see Minimal Reproducible Example below)*. I also tried using a different plugin such as [rollup-plugin-csv](rollup-plugin-csv) and got the same result. 

I tried it on a basic vite installation and everything works fine.

Here's an example:
![image](https://user-images.githubusercontent.com/28738855/146694679-5287bf28-c249-4f1d-9429-1b3abc1f38fd.png)

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-rpfxhk-maaecr?file=src%2Fpages%2Findex.astro&on=stackblitz

**Comment by matthewp** [comment]
Probably going to need some help figuring this one out. I'm not sure what Astro could possibly be doing here.

**Comment by mattrothenberg** [comment]
@matthewp I'm seeing this as well! Curious if you have any leads or ideas as to what might be going on

**Comment by mattrothenberg** [comment]
@matthewp Alright so I did some digging and found a few interesting data points.

https://github.com/withastro/astro/blob/main/packages/astro/src/vite-plugin-astro/index.ts#L22

In this file, I noticed that if I deleted `config.vite` at the top of the `astro` function, CSV imports started working perfectly.

```js
*******
// this is the value of 'config.vite'
{ plugins: [ { name: 'dsv', transform: [Function: transform] } ] }
*******
```

Do you have any idea why the mere presence of this object (with the DSV plugin) would cause the CSV import to result in the weird scenario above (with `export default` being tacked on to the beginning of each row as a string)

**Comment by FredKSchott** [comment]
Fixed in v0.24.0!
